src/Awaitable.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Interop\Async;
+
+/**
+ * Simple interface for awaitable objects representing the future value of asynchronous operations.
+ */
+interface Awaitable
+{
+    /**
+     * Registers a callback to be invoked when the awaitable is resolved.
+     *
+     * @param callable(\Throwable|\Exception $exception = null, mixed $result = null) $onResolved
```

Comment:
Yeah, that's better since the function will always be provided both arguments. Someday we can update it to `callable(?\Throwable $exception, mixed $result)` :smile_cat: